tests/sources/xml/test_oai_dc.py

```diff
@@ -54,26 +65,217 @@ def test_oaidctransform_with_all_fields_transforms_correctly():
+        OaiDc("", oai_dc_record_all_fields).get_content_type()
+
+
+def test_get_content_type_raises_key_error_if_source_missing(oai_dc_record_all_fields):
+    with pytest.raises(KeyError):
+        OaiDc(None, oai_dc_record_all_fields).get_content_type()
+
+
+def test_get_contributors_success():
+    source_record = create_oai_dc_source_record_stub(
+        """
+        <dc:creator>Ye Li</dc:creator>
+        """
+    )
+    assert OaiDc.get_contributors(source_record) == [
+        timdex.Contributor(
+            value="Ye Li",
+            affiliation=None,
+            identifier=None,
+            kind="Creator",
+            mit_affiliated=None,
+        )
+    ]
+
+
+def test_get_contributors_transforms_correctly_if_fields_blank(
+    oai_dc_record_optional_fields_blank,
+):
+    assert OaiDc.get_contributors(oai_dc_record_optional_fields_blank) == []
+
+
+def test_get_contributors_transforms_correctly_if_fields_missing(
+    oai_dc_record_optional_fields_missing,
+):
+    assert OaiDc.get_contributors(oai_dc_record_optional_fields_missing) == []
+
+
+def test_get_dates_success():
+    source_record = create_oai_dc_source_record_stub(
+        """
+        <dc:date>2008-06-19T17:55:27</dc:date>
+        """
+    )
+    assert OaiDc.get_dates(source_record_id="abc", source_record=source_record) == [
+        timdex.Date(kind="Unknown", value="2008-06-19T17:55:27")
     ]
+
+
+def test_get_dates_transforms_correctly_if_fields_blank(
+    oai_dc_record_optional_fields_blank,
+):
+    assert (
+        OaiDc.get_dates(
+            source_record_id="abc", source_record=oai_dc_record_optional_fields_blank
+        )
+        == []
+    )
+
+
+def test_get_dates_transforms_correctly_if_fields_missing(
+    oai_dc_record_optional_fields_missing,
+):
+    assert (
+        OaiDc.get_dates(
+            source_record_id="abc", source_record=oai_dc_record_optional_fields_missing
+        )
+        == []
+    )
+
+
+def test_get_dates_transforms_correctly_if_date_invalid():
+    source_record = create_oai_dc_source_record_stub(
+        """
+        <dc:date>INVALID</dc:date>
+        """
+    )
+    assert OaiDc.get_dates(source_record_id="abc", source_record=source_record) == []
+
+
+def test_get_identifiers_success(oai_dc_record_all_fields):
+    assert OaiDc.get_identifiers(oai_dc_record_all_fields) == [
+        timdex.Identifier(value="oai:libguides.com:guides/175846", kind="OAI-PMH")
+    ]
+
+
+def test_get_identifiers_transforms_correctly_if_fields_blank(
+    oai_dc_record_optional_fields_blank,
+):
+    oai_dc_record_optional_fields_blank.header.identifier.clear()
+    assert OaiDc.get_identifiers(oai_dc_record_optional_fields_blank) == []
+
+
+def test_get_identifiers_transforms_correctly_if_fields_missing(
+    oai_dc_record_optional_fields_missing,
+):
+    oai_dc_record_optional_fields_missing.header.identifier.decompose()
+    assert OaiDc.get_identifiers(oai_dc_record_optional_fields_missing) == []
```

Comment:
Curious what you think of this setup. 🤔 The identifiers are derived from a child of the `header` element, so using the stub (as it is written) or the `oai_dc_record_*` fixtures directly was not possible.

Reply:
I think that works

Reply:
Update from @ghukill : Whole `header` element should be overwritten.